salt/utils/process.py

```diff
@@ -755,12 +721,17 @@ def __getstate__(self):
         del self._args_for_getstate
         del self._kwargs_for_getstate
         return {'args': args,
-                'kwargs': kwargs}
+                'kwargs': kwargs,
+                '_after_fork_methods': self._after_fork_methods,
+                '_finalize_methods': self._finalize_methods,
+                }
 
-    def __setup_process_logging(self):
+    def _setup_process_logging(self):
         salt.log.setup.setup_multiprocessing_logging(self.log_queue)
 
     def _run(self):
+        for method, args, kwargs in self._after_fork_methods:
+            method(*args, **kwargs)
```

Comment:
No need to wrap in a try/except to make sure all run?

Reply:
Since this api is internal to the class only I think we're safer to fail fast and let the exception get raised if there is one. We'll be more likely see and fix any issues that come up.

Reply:
Ok